haystack/nodes/retriever/_embedding_encoder.py

```diff
@@ -434,11 +438,93 @@ def save(self, save_dir: Union[Path, str]):
         raise NotImplementedError(f"Saving is not implemented for {self.__class__}")
 
 
+class _BedrockEmbeddingEncoder(_BaseEmbeddingEncoder):
```

Comment:
Added cohere model support in addition to titan, also initialized client using session. Added profile_name.